apps/examples/smartfs/fs_performance/fs_performance_main.c

```diff
@@ -0,0 +1,522 @@
+	int frt_fd = -1;
+
+	memset(g_rw_buf, '0', MAX_BUF_SIZE);
+	frt_fd = open(g_timer_path, O_RDONLY);
+	if (frt_fd < 0) {
+		printf("Unable to open file: %s errno: %d\n", g_timer_path, errno);
+		return ERROR;
+	}
+
+	if (ioctl(frt_fd, TCIOC_SETMODE, MODE_FREERUN) < 0) {
+		fprintf(stderr, "ERROR: Failed to set Free Run Timer: %d\n", errno);
+		ret = -1;
+		goto close_with_frt;
+	}
+
+	if (ioctl(frt_fd, TCIOC_START, TRUE) < 0) {
+		fprintf(stderr, "ERROR: Failed to start Free Run Timer: %d\n", errno);
+		ret = -1;
+		goto close_with_frt;
+	}
+
+	switch (option) {
+
+	case TEST_FLAG_FILE_CREATION :
+		printf("\n****************************************************************************************\n");
+		printf("********************************TEST FILE CREATION**********************************");
+		printf("\n****************************************************************************************\n");
+		printf("File size: %d bytes\n", g_fsz);
+
+		if (ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&before) < 0) {
+			fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+			ret = -1;
+			goto close_with_frt;
+		}
+
+		fd = open(g_file[0], O_WROK | O_CREAT);
+		if (fd < 0) {
+			printf("Unable to open file: %s, fd: %d\n", g_file[0], fd);
+			ret = ERROR;
+			goto close_with_frt;
+		}
+
+		ret = write(fd, g_rw_buf, g_fsz);
+		if (ret != (g_fsz)) {
+			printf("Unable to write in file: %s, size: %d, ret: %d\n", g_file[0], g_fsz, ret);
+			ret = ERROR;
+			goto close_with_fd;
+		}
+
+		close(fd);
+
+		if (ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&after) < 0) {
+			fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+			ret = -1;
+			goto close_with_frt;
+		}
+
+		tot_time  = after.timeleft - before.timeleft;
+		printf("Total time taken is: %lu us\n", tot_time);
+
+		ret = unlink(g_file[0]);
+		if (ret != OK) {
+			printf("Failed to unlink file %s, errno : %d\n", g_file[0], errno);
+			ret = ERROR;
+			goto close_with_frt;
+		}
+		break;
+
+	case TEST_FLAG_OVERWRITE :
+		printf("\n****************************************************************************************\n");
+		printf("**********************************TEST OVERWRITE************************************");
+		printf("\n****************************************************************************************\n");
+		printf("File size: %d bytes, Buffer size: %d bytes\n", g_fsz, g_bsz);
+
+		fd = open(g_file[0], O_WROK | O_CREAT);
+		if (fd < 0) {
+			printf("Unable to open file: %s, fd: %d\n", g_file[0], fd);
+			ret = ERROR;
+			goto close_with_frt;
+		}
+
+		for (i = 0; i < (g_fsz / g_bsz); i++) {
+			ret = write(fd, g_rw_buf, g_bsz);
+			if (ret != (g_bsz)) {
+				printf("Unable to write in file: %s, ret: %d\n", g_file[0], ret);
+				ret = ERROR;
+				goto close_with_fd;
+			}
+		}
+		close(fd);
+
+		for (i = 1; i <= ITR_OVE; i++) {
+			if (ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&before) < 0) {
```

Comment:
please apply below style (not only ioctl but also other function)
```
ret = ioctl(....);
if (ret != OK) {
    goto some_where;
}
....
    return OK;
some_where : 
    do_something;
    if (ret != OK) {
          return ERROR;
    }
    do_something;
```

Reply:
Done.